apex_launchtest/apex_launchtest/apex_runner.py

```diff
@@ -177,20 +197,40 @@ def _print_process_output_summary(self):
                 print("{}".format(io.text.decode('ascii')))
             print("#" * (len(process.process_name) + 21))
 
+    def _bind_test_context_to_tests(self, context, test_suite):
+        # Look for tests that expect additional arguments and bind items from the context
+        # to the tests
+        for test in ApexRunner._iterate_tests_in_test_suite(test_suite):
+            # Need to reach a little deep into the implementation here to get the test
+            # method.  See unittest.TestCase
+            test_method = getattr(test, test._testMethodName)
+
+            # We only want to bind the part of the context matches the test args
+            test_arg_names = inspect.getfullargspec(test_method).args
+            matching_args = {k: v for (k, v) in context.items() if k in test_arg_names}
+
+            # Replace the test with a functools.partial that has the arguments
+            # provided by the test context already bound
+            setattr(test, test._testMethodName, functools.partial(test_method, **matching_args))
```

Comment:
In a past life, this would've been a bad code smell to me, but other test frameworks do similar things to bind arguments.

Reply:
Yeah, I'd imagine `pytest` does something alike to bind its fixture objects as test functions' arguments.